File: apps/api/src/modules/summary/summary.controller.ts
import { Controller, Get, Inject, Param } from "@nestjs/common";
import type { DaySummaryDto } from "@today/contracts";
import { SummaryService } from "./summary.service";

@Controller("v1/summaries")
export class SummaryController {
  constructor(
    @Inject(SummaryService) private readonly summaries: SummaryService,
  ) {}

  @Get(":date")
  getByDate(@Param("date") date: string): DaySummaryDto {
    return this.summaries.getByDate(date);
  }
}

File: apps/api/src/main/java/com/today/summary/SummaryController.java
package com.today.summary;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/v1/summaries")
public class SummaryController {

  private final SummaryService summaries;

  public SummaryController(SummaryService summaries) {
    this.summaries = summaries;
  }

  @GetMapping("/{date}")
  public DaySummaryDto getByDate(@PathVariable String date) {
    return summaries.getByDate(date);
  }
}
